src/test/func/memcpy/func-memcpy.cc

```diff
@@ -144,6 +160,49 @@ void check_bounds(size_t size, size_t out_of_bounds)
   my_free(d);
 }
 
+void check_overlaps1()
+{
+  size_t size = 16;
+  START_TEST("memmove overlaps1");
+  auto* s = static_cast<unsigned int*>(my_malloc(size * sizeof(unsigned int)));
+  for (size_t i = 0; i < size; ++i)
+  {
+    s[i] = static_cast<unsigned int>(i);
+  }
+  my_memmove(&s[2], &s[4], sizeof(s[0]));
+  EXPECT(s[2] == s[4], "overlap error: {} {}", s[2], s[4]);
+  my_memmove(&s[15], &s[5], sizeof(s[0]));
+  EXPECT(s[15] == s[5], "overlap error: {} {}", s[15], s[5]);
+  auto ptr = s;
+  my_memmove(ptr, s, size * sizeof(s[0]));
+  EXPECT(ptr == s, "overlap error: {} {}", ptr, s);
+  my_free(s);
+}
+
+template<bool after>
+void check_overlaps2(size_t size)
+{
+  START_TEST("memmove overlaps2, size {}", size);
+  auto sz = size / 2;
```

Comment:
I wonder if it would be good to separate the offset from the size being copied. The current test only covers very small overlaps.
```suggestion
  auto sz = size / 2;
  auto offset = size / 3;
```
Then use `offset` in the calculations using `after`?